File: front/src/app/models/chatRoom.ts
import { Message } from "./message";

export class ChatRoom {


    constructor(id:number,userA:number,userB:number,messages:Message[]){
        this.id = id;
        this.userA = userA;
        this.userB = userB;
        this.messages = messages;
    }

    id:number;
    userA:number;
    userB:number;
    messages:Message[];

}

File: back/src/main/java/fr/mdazin/chatter/model/ChatRoom.java
package fr.mdazin.chatter.model;


import lombok.*;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class ChatRoom {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private Integer userA;

    private Integer userB;

    @OneToMany(mappedBy="chatRoom",fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Message> messages;



}
